components/salsa-2022-macros/src/tracked_fn.rs

```diff
@@ -85,6 +71,206 @@ impl crate::options::AllowedOptions for TrackedFn {
     const CONSTRUCTOR_NAME: bool = false;
 }
 
+type ImplArgs = Options<TrackedImpl>;
+
+pub(crate) fn tracked_impl(
+    args: proc_macro::TokenStream,
+    mut item_impl: syn::ItemImpl,
+) -> syn::Result<TokenStream> {
+    let args: ImplArgs = syn::parse(args)?;
+    let self_type = match &*item_impl.self_ty {
+        syn::Type::Path(path) => path,
+        _ => {
+            return Err(syn::Error::new(
+                item_impl.self_ty.span(),
+                "#[salsa::tracked] can only be applied to salsa structs",
+            ))
+        }
+    };
+    let extra_impls = item_impl
+        .items
+        .iter_mut()
+        .filter_map(|item| {
+            let item_method = match item {
+                syn::ImplItem::Method(item_method) => item_method,
+                _ => return None,
+            };
+            let salsa_tracked_attr = item_method.attrs.iter().position(|attr| {
+                attr.path
+                    .get_ident()
+                    .map(|ident| ident == "tracked")
+                    .unwrap_or(false)
+            })?;
```

Comment:
Can we detect `salsa::tracked` here instead? 

Maybe we just use that for consistency?

Alternatively, we could give a better error that way.

Reply:
Changed to #[salsa::tracked] and added an error message for if you don't put #[salsa::tracked] on the impl block.

Reply:
Great! Can you add a test in the `salsa-2022-tests/tests/compile-fail` directory to show the error cases? You can use the [trybuild workflow](https://github.com/dtolnay/trybuild#workflow) to generate the stderr files.

Reply:
(request: add an error test into salsa-2022-tests/tests/compile-fail)

Reply:
I've added a simple test for the error message I added.